climada/hazard/base.py

```diff
@@ -1541,6 +1543,48 @@ def local_exceedance_inten(self, return_periods=(25, 50, 100, 250)):
                    Reason: no negative intensity values were found in hazard.')
             inten_stats[inten_stats < 0] = 0
         return inten_stats
+    
+    def local_return_period(self, hazard_intensities):
+        """Compute local return periods for given hazard intensities.
+    
+        Parameters
+        ----------
+        hazard_intensities : np.array
+            Hazard intensities to consider.
```

Comment:
Be careful, this might lead to very large memory usage. Hazard intensities are on purpose sparse matrices.

Reply:
See comment above, hazard_intensities usually only includes a few values.